qemu/tests/balloon_check.py

```diff
@@ -91,6 +92,54 @@ def memory_check(self, step, ballooned_mem):
                 raise exceptions.TestFail("Balloon test failed %s" % step)
         return (mmem, gmem)
 
+    def memory_stats_check(self, keyname):
+        """
+        Check memory statistics via balloon service.
+
+        :param keyname: key name of the output of the 'qom-get' property.
+        :return: if memory statistics can be got via balloon service, reture True,
+                 else return False.
+        :rtype: bool
+        """
+        check_mem_ratio = float(self.params.get("check_mem_ratio", 0.1))
+        sleep_time = int(self.params.get("polling_sleep_time", 20))
+        base_path = self.params.get("base_path", "/machine/peripheral/")
+        device = self.params.get("balloon", "balloon0")
+        device_path = os.path.join(base_path, device)
+        polling_interval = int(self.params.get("polling_interval", 2))
+        set_balloon_property = self.params.get("set_balloon_property",
+                                               "guest-stats-polling-interval")
+        get_balloon_property = self.params.get("get_balloon_property", "guest-stats")
```

Comment:
This is incorrect, you may set/get different properties in one test, so the property name cannot be set by one single param, the right way to handle this thing is making them as enum type (for me, string type is also acceptable) and then pass them to functions, for example:
```python
BLN_PROP_POLLING_INTERVAL = "guest-stats-polling-interval"
BLN_PROP_STATS = "guest-stats"
...

def set_balloon_property(key, value):
    ...

def get_balloon_property(key):
    ...

set_balloon_property(BLN_PROP_POLLING_INTERVAL, value)
value = get_balloon_property(BLN_PROP_STATS)
```

Reply:
We do not need any other properties in all balloon service tests refer to https://github.com/qemu/qemu/blob/master/docs/virtio-balloon-stats.txt